`ColorPicker` remove unused children and fix sample bugs.

WhalesState opened the pull request:
* Fixes: https://github.com/godotengine/godot/issues/88073

Remove an unused `VBoxContainer`.
Remove a redundant `VBoxContainer` and add it's children directly to main vbox.
Remove 3 unused children from grid (Label, Slider, SpinBox).
Fix current color overbright position.
Added color mode buttons and sliders to a another `VBoxContainer` to hide both when `sliders_visible` is false.
Use `tab_hovered` instead of `tab_pressed` when hovering a mode button.

KoBeWi commented:
Pretty sure changing base type breaks compatibility, which is why the redundant VBoxContainer had to exist.

WhalesState commented:
> Pretty sure changing base type breaks compatibility, which is why the redundant VBoxContainer had to exist.

What is the use case of the `MarginContainer` ? if for the color button popup, then why don't we include the margins inside the popup's panel ?

WhalesState commented:
Reverted the base class to `VBoxContainer` again, it's easier to be reviewed now.

KoBeWi commented:
> What is the use case of the MarginContainer?

The use case is that ColorPicker doesn't need to be used with popup. It's a stand-alone node, ColorPickerButton only wraps it.
The container was added in #78468

WhalesState commented:
Check the title, should i open a new issue for each fix to track them?

KoBeWi commented on scene/gui/color_picker.cpp:
```diff
@@ -1258,13 +1258,15 @@ void ColorPicker::_update_text_value() {
-			// Revert to the old color when left-clicking the old color sample.
-			set_pick_color(old_color);
-			emit_signal(SNAME("color_changed"), color);
+	if (display_old_color) {
```
```suggestion
	if (!display_old_color) {
		return;
	}
```
Less nesting and less changes.

KoBeWi commented on scene/gui/color_picker.cpp:
```diff
@@ -2234,8 +2236,11 @@ ColorPicker::ColorPicker() {
+
 	mode_hbc = memnew(HBoxContainer);
-	real_vbox->add_child(mode_hbc);
+	sliders_vbox->add_child(mode_hbc);
```
This changes the behavior of `sliders_visible` to also hide color modes. Why this change?
I know color modes are *probably* not useful without sliders, but they are still 2 separate properties.

KoBeWi commented:
There is a behavior change, which is questionable. Everything else looks fine.

KoBeWi commented on scene/gui/color_picker.cpp:
```diff
@@ -2177,7 +2179,7 @@ void ColorPicker::_bind_methods() {
 	BIND_THEME_ITEM_EXT(Theme::DATA_TYPE_STYLEBOX, ColorPicker, mode_button_normal, "tab_unselected", "TabContainer");
 	BIND_THEME_ITEM_EXT(Theme::DATA_TYPE_STYLEBOX, ColorPicker, mode_button_pressed, "tab_selected", "TabContainer");
-	BIND_THEME_ITEM_EXT(Theme::DATA_TYPE_STYLEBOX, ColorPicker, mode_button_hover, "tab_selected", "TabContainer");
+	BIND_THEME_ITEM_EXT(Theme::DATA_TYPE_STYLEBOX, ColorPicker, mode_button_hover, "tab_hovered", "TabContainer");
```
Not important, but this style has visible borders at sides:
![image](https://github.com/user-attachments/assets/c5b35f49-6e85-4ef2-b74b-162548a63801)

WhalesState replied:
Also the `tab_unselected` has the same left and right borders but it's less visible, was thinking about removing those borders but less visible.

They are clear here.
![image](https://github.com/user-attachments/assets/c76fd6e6-8843-4f9e-acc4-46d599f19eb7)

WhalesState committed:
`ColorPicker` remove unused children and fix sample bugs.

Remove an unused `VBoxContainer`.
Remove 3 unused children from grid (Label, Slider, SpinBox).
Ignore sample input when old color is not displayed.
Fix current color overbright position.

WhalesState commented:
Applied all the suggestions.

Repiteo commented:
Thanks!
